### ISSUE TITLE
Solr trys to create own log-folder in the store

### ISSUE THREAD

**Issue by Mirroregg123** [NixOS/nixpkgs#81732]
## Issue description

I just installed solr and tried to start it with
`solr start`
it results in the following error 
`ERROR: Logs directory /nix/store/h4ign4lplc6l9nw6jrdk6zz57ijkilsl-solr-8.3.0/server/logs could not be created. Exiting`

## Technical details

I have the following system settings
```
 - system: `"x86_64-linux"`
 - host os: `Linux 5.4.17, NixOS, 19.09.2036.c49da6435f3 (Loris)`
 - multi-user?: `no`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.3.2`
 - channels(root): `"nixos-19.09.2036.c49da6435f3"`
 - channels(melr): `""`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
```
and the following written in my configuration.nix
```
  services.solr = {
    enable = true;
    stateDir = "/var/lib/solr";
    port = 5555;
  };
```

When I have a look at the solar.service, I find also the following
```
Environment="SOLR_HOME=/var/lib/solr/data"
Environment="SOLR_LOGS_DIR=/var/lib/solr/logs"
Environment="SOLR_PORT=5555"
```

After typing the configuration the configuration.nix and rebuilding the system, the folder `/var/lib/solr` was created successfully

## Question

Can anyone tell me, where my misstake is?
I also tried the examples on [https://lucene.apache.org/solr/guide/8_4/solr-tutorial.html](url)
The error is always the same.

**Comment by aanderse** [comment]
Generally my thought would be to manage this via `systemd` but yeah this is a bug that should be remedied likely by wrapping the exposed `solr` binary with these environment variables.

Thanks for reporting this and consider me on it :+1:

**Comment by aanderse** [comment]
@Mirroregg123 a key concept to nixos that you might have overlooked is that we split packages from services, where most other gnu/linux distros don't separate the two concepts. If you want to run a `solr` server on nixos you shouldn't install the package, instead you should enable the service (via `services.solr.enable`, as you found out). Generally our services are managed by `systemd` so you can do things like `systemctl stop solr.service` and `systemctl start solr.service`, if you need to.

I put together a quick hack to expose the required environment variables to `solr` in [this](https://github.com/aanderse/nixpkgs/tree/solr) branch, but I'm not convinced it is a great solution... nor have I really tested it.

Does the explanation of services being separate from packages make the situation any clearer?

**Comment by Mirroregg123** [comment]
@aanderse 
I "installed" solr indeed as a service.
After enabling the service with the written options, I'm able to execute `solr start`, with the mentioned error, but I understand, that I might don't want to do this at all, rather then working with systemd.

The current situation is that I tried `systemctl start solr.service` and now I have a web-application at [http://localhost:5555](url) 

So this is great. Still I'm a bit puzzled by the following.

In the web application I'm not able to create a new core. Hence I tried (successfully) doing this on the terminal with
```
sudo -u solr solr create_core -c new_core
```

Successfully means, that I have now a folder `/var/lib/solr/data/new_core`. And the core is also displayed in the web application. 
(Short remark: just `solr create_core -c new_core` failed with `ERROR: Failed to create new core instance directory: /var/lib/solr/data/new_core`)

So in this case `solr` really works on the specified path.

For my current task I'm satisfied with the situation and close the ticket. Thanks for the assist.
